app/common/rational.cpp

```diff
@@ -3,421 +3,312 @@
 
 #include "rational.h"
 
+#include <cmath>
+
+using namespace std;
+
+rational::rational(const intType& num)
+    : numer(num)
+      , denom(1)
+{}
+
+rational::rational(const intType& num, const intType& den)
+    : numer(num)
+      , denom(den)
+{
+    fixSigns();
+    reduce();
+}
 rational::rational(const AVRational &r) :
-  numer(r.num),
-  denom(r.den)
+    numer(r.num),
+    denom(r.den)
 {
 }
 
 //Function: print number to cout
 
 void rational::print(ostream &out) const
 {
-  out << this->numer << "/" << this->denom;
+    out << this->numer << "/" << this->denom;
 }
 
 //Function: ensures denom >= 0
 
 void rational::fixSigns()
 {
-  if(denom < 0)
-    {
-      denom = -denom;
-      numer = -numer;
-    }
-  if(numer == intType(0) || denom == intType(0))
-    {
-      numer = intType(0);
-      denom = intType(0);
-    }
+    numer *= 1 - 2 * std::signbit(denom);
+    denom = std::abs(denom);
 }
 
 //Function: ensures lowest form
 
 void rational::reduce()
 {
-  // Euclidean often fails if numbers are negative, we abs it and re-neg it later if necessary
-  bool neg = numer < 0;
-
-  numer = qAbs(numer);
-
-  intType d = 1;
-
-  if(denom != 0 && numer !=0)
-    d = gcd(numer, denom);
-
-  if(d > 1)
-    {
-      numer /= d;
-      denom /= d;
-    }
-
-  if (neg) {
-    numer = -numer;
-  }
+    intType d = gcd(std::abs(numer), denom);
+    numer /= d;
+    denom /= d;
```

Comment:
mmm... just reviewing my own changes... this might fail in case ``denom == 0``
this needs an ``if``